renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -1938,14 +1944,27 @@ bool WrappedVulkan::EndFrameCapture(void *dev, void *wnd)
 
       const SwapchainInfo &swapInfo = *swaprecord->swapInfo;
 
-      presentInfo = &swapInfo.lastPresent;
-      backbuffer = swapInfo.images[presentInfo->imageIndex].im;
+      backbuffer = swapInfo.images[swapInfo.lastPresent.imageIndex].im;
+      swapImageInfo = &swapInfo.imageInfo;
+      swapQueueIndex = GetRecord(swapInfo.lastPresent.presentQueue)->queueFamilyIndex;
+      swapLayout =
+          swapInfo.shared ? VK_IMAGE_LAYOUT_SHARED_PRESENT_KHR : VK_IMAGE_LAYOUT_PRESENT_SRC_KHR;
 
       // mark all images referenced as well
       for(size_t i = 0; i < swapInfo.images.size(); i++)
         GetResourceManager()->MarkResourceFrameReferenced(GetResID(swapInfo.images[i].im),
                                                           eFrameRef_Read);
     }
+    else if(VRBackbufferRecord)
+    {
+      RDCASSERT(VRBackbufferRecord->resInfo);
+      backbuffer = GetResourceManager()->GetCurrentHandle<VkImage>(m_CurrentVRBackbuffer);
+      swapImageInfo = &VRBackbufferRecord->resInfo->imageInfo;
+      swapQueueIndex = m_QueueFamilyIdx;
+      swapLayout = VK_IMAGE_LAYOUT_READ_ONLY_OPTIMAL;
```

Comment:
Do we need to feature-guard this use of `VK_IMAGE_LAYOUT_READ_ONLY_OPTIMAL`? Or are we saying as part of the 'API' contract that this UUID will only be passed through when `VK_KHR_synchronization2` is available and enabled to allow use of the image layout?

Reply:
Ah damn, too many layouts ^^. 
I meant VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL. The API contract in XR is that apps will switch the rendertargets to VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL before xr_endframe (so that the VR compositor can read it)